test/unit/request-wrapper.test.js

```diff
@@ -801,22 +878,62 @@ describe('formatError', () => {
     // save the original code
     const originalCode = basicAxiosError.code;
 
-    basicAxiosError.code = 'DEPTH_ZERO_SELF_SIGNED_CERT';
-    const error = requestWrapperInstance.formatError(basicAxiosError);
+    axiosErrorCopy.code = 'DEPTH_ZERO_SELF_SIGNED_CERT';
+    const error = requestWrapperInstance.formatError(axiosErrorCopy);
 
     // put the original message back in, before expectations in case they fail
-    basicAxiosError.code = originalCode;
+    axiosErrorCopy.code = originalCode;
 
     expect(error instanceof Error).toBe(true);
     expect(error.message).toMatch(/SSL certificate is not valid/);
   });
 
   it('check the message flow', () => {
-    delete basicAxiosError.request;
-    const error = requestWrapperInstance.formatError(basicAxiosError);
+    delete axiosErrorCopy.request;
+    const error = requestWrapperInstance.formatError(axiosErrorCopy);
     expect(error instanceof Error).toBe(true);
     expect(error.message).toBe('error in building the request');
   });
+
+  it('should convert string response bodies to objects if content is JSON', () => {
+    const newAxiosError = makeCopy(basicAxiosError);
+    newAxiosError.response.data = '{ "errorMessage": "some error" }';
+    expect(newAxiosError.response.headers['content-type']).toBe('application/json');
```

Comment:
let's try a content-type that has extra stuff (e.e. `application/json; charset=utf8`.